src/librustc/infer/error_reporting/nice_region_error/named_anon_conflict.rs

```diff
@@ -110,9 +110,10 @@ impl<'a, 'gcx, 'tcx> NiceRegionError<'a, 'gcx, 'tcx> {
             E0621,
             "explicit lifetime required in {}",
             error_var
-        ).span_label(
+        ).span_suggestion(
             arg.pat.span,
-            format!("consider changing {} to `{}`", span_label_var, new_ty),
+            &format!("consider changing {} to `{}`", span_label_var, new_ty),
+            "".to_string()
```

Comment:
When applying the suggestion, the code will be incorrect. The fourth argument should the `new_ty`, as that's what the code needs to be after applying the change, and the span needs to be changed to point at the argument type, instead of the argument name so the suggestion gets applied in the right place.

Reply:
@estebank make sense! I will fix this by today or tomorrow.

Reply:
Friendly ping :)

Reply:
@estebank I think I have found the way to find span of argument type. doing experiements now.

Reply:
@estebank: done :)